source/val/validate_ray_query.cpp

```diff
@@ -23,6 +23,15 @@ namespace spvtools {
 namespace val {
 namespace {
 
+uint32_t GetArrayLength(ValidationState_t& _, const Instruction* array_type) {
+  assert(array_type->opcode() == spv::Op::OpTypeArray);
+  uint32_t const_int_id = array_type->GetOperandAs<uint32_t>(2U);
+  Instruction* array_length_inst = _.FindDef(const_int_id);
+  assert(array_length_inst->opcode() == spv::Op::OpConstant);
```

Comment:
Arrays could be sized via a specialization constant so this assertion seems too strong. Maybe it should be an error if the length is not OpConstant, but the spec isn't clear whether OpSpecConstant of length 2 would be valid.